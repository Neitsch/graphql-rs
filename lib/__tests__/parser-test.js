const graphql_rs = require("../index");
const graphql = require("graphql");

describe("parses graphql", () => {
  it("simple", () => {
    expect(
      graphql_rs.parse("type User { name: String id: ID }")
    ).toMatchSnapshot();
  });
  xit("compare", () => {
    const sourceText = "type User { name: String id: ID }";
    expect(graphql_rs.parse(sourceText)).toEqual(graphql.parse(sourceText));
  });
});
