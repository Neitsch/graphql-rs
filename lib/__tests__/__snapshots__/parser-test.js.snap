// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses graphql simple 1`] = `
Object {
  "definitions": Array [
    Object {
      "description": null,
      "directives": null,
      "fields": Array [
        Object {
          "arguments": null,
          "description": null,
          "directives": null,
          "kind": "FieldDefinition",
          "loc": null,
          "name": Object {
            "kind": "Name",
            "loc": null,
            "value": "name",
          },
          "type": Object {
            "kind": "NamedType",
            "loc": null,
            "name": Object {
              "kind": "Name",
              "loc": null,
              "value": "String",
            },
          },
        },
        Object {
          "arguments": null,
          "description": null,
          "directives": null,
          "kind": "FieldDefinition",
          "loc": null,
          "name": Object {
            "kind": "Name",
            "loc": null,
            "value": "id",
          },
          "type": Object {
            "kind": "NamedType",
            "loc": null,
            "name": Object {
              "kind": "Name",
              "loc": null,
              "value": "ID",
            },
          },
        },
      ],
      "interfaces": null,
      "kind": "ObjectTypeDefinition",
      "loc": null,
      "name": Object {
        "kind": "Name",
        "loc": null,
        "value": "User",
      },
    },
  ],
  "kind": "Document",
  "loc": null,
}
`;
